tests/integration/test_subscriptions_api.py

```diff
@@ -1,59 +1,143 @@
+    all_subs = [{'id': str(i), 'status': 'created'} for i in range(1, 101)]
+    select_subs = (sub for sub in all_subs
+                   if not status or sub['status'] in status)
+    pages = []
+    for group in zip_longest(*[iter(select_subs)] * size):
+        subs = list(sub for sub in group if sub is not None)
+        page = {'subscriptions': subs, '_links': {}}
+        pm = datetime.now().isoformat()
+        if len(subs) == size:
+            page['_links'][
+                'next'] = f'https://api.planet.com/subscriptions/v1?pm={pm}'
+        pages.append(page)
+    return pages
+
+
+# By default, respx's mock router asserts that all configured routes
+# are followed in a test. Each of our parameterized tests follows only
+# one route, so if we want to configure routes once and reuse them, we
+# must disable the default.
+api_mock = respx.mock(assert_all_called=False)
+
+# 1. Request for status: created. Response has three pages.
+api_mock.route(M(url__startswith='https://api.planet.com/subscriptions/v1'),
+               M(params__contains={'status': 'created'})).mock(side_effect=[
+                   Response(200, json=page)
+                   for page in sub_pages(status={'created'}, size=40)
+               ])
+
+# 2. Request for status: failed. Response has a single empty page.
+api_mock.route(M(url__startswith='https://api.planet.com/subscriptions/v1'),
+               M(params__contains={'status': 'failed'})).mock(
+                   side_effect=[Response(200, json={'subscriptions': []})])
+
+# 3. Request for status: queued. Response has a single empty page.
+api_mock.route(M(url__startswith='https://api.planet.com/subscriptions/v1'),
+               M(params__contains={'status': 'queued'})).mock(
+                   side_effect=[Response(200, json={'subscriptions': []})])
+
+# 4. No status requested. Response is the same as for 1.
+api_mock.route(
+    M(url__startswith='https://api.planet.com/subscriptions/v1')).mock(
+        side_effect=[Response(200, json=page) for page in sub_pages(size=40)])
+
+
+# This mock API returns posted data to the caller. It's used to test
+# methods that rely on POST or PUT.
+def modify_response(request):
+    if request.content:
+        return Response(200, json=json.loads(request.content))
+    else:
+        return Response(200)
+
+
+modify_api_mock = respx.mock()
+modify_api_mock.route(
+    M(url__startswith='https://api.planet.com/subscriptions/v1'),
+    M(method__in=['POST', 'PUT'])).mock(side_effect=modify_response)
+
+sub_api_mock = respx.mock()
+sub_api_mock.route(
+    M(url__startswith='https://api.planet.com/subscriptions/v1'),
+    M(method__in=['GET'])).mock(
+        return_value=Response(200,
+                              json={
+                                  'id': '42',
+                                  'name': 'test',
+                                  'delivery': 'yes, please',
+                                  'source': 'test'
+                              }))
```

Comment:
This is most of the work: creating the mock router configurations that allow us to test features.